src/hooks/useSAMLSignIn.ts

```diff
@@ -12,9 +39,28 @@ export const useSAMLSignIn = (samlAuthProviderId?: string) => {
       samlAuthProviderId
     );
 
+    if (showLoginLoading) {
+      showLoginLoading();
+    }
```

Comment:
If we just use `useAsync` instead of `useCallback` we'll have proper loading status built in for free.

Reply:
Removed in https://github.com/sparkletown/sparkle/pull/1667/commits/c77cd255fb597da0dd2611bc66089e77dbe0e59f but you can't use `useAsync` there since it's happening inside the function

Reply:
> but you can't use `useAsync` there since it's happening inside the function

What do you mean by that?

There are 3 different variations of `useAsync`, do none of them work here?

- https://github.com/streamich/react-use/blob/master/docs/useAsync.md
- https://github.com/streamich/react-use/blob/master/docs/useAsyncFn.md
- https://github.com/streamich/react-use/blob/master/docs/useAsyncRetry.md

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1667/commits/36f9ee2f4cfc1f5342c9ef7c0843ba5452393d9f